lib/common/common/src/counter/hardware_accumulator.rs

```diff
@@ -47,3 +101,46 @@ impl Default for HwMeasurementAcc {
         }
     }
 }
+
+/// A type collecting hardware measuerements in multi threaded scenarios where multiple instances of `HwMeasurementAcc` would be needed.
+/// Because `HwMeasurementAcc` purposely doesn't implement Clone, we need this type to allow having an owned type we can move
+/// between threads/async closures.
+///
+/// This collector automatically applies all measurements to the `HwMeasurementAcc` that was used when creating this collector on drop.
+/// To ensure all values get applied, before reading from a `HwMeasurementAcc`, this type must be dropped first.
+pub struct HwMeasurementCollector {
+    /// Shared reference of CPU usage with the given Parent.
+    cpu_counter: Arc<AtomicUsize>,
+
+    /// A new temporary measurement accumulator, which values get applied to the parents
+    /// ones on drop.
+    collector: HwMeasurementAcc,
```

Comment:
I like this idea, but now we have a circular dependency between structures. Both of `HwMeasurementCollector` and `HwMeasurementAcc` require each other. I would prefer to avoid that.